xgboost_ray/main.py

```diff
@@ -321,6 +321,24 @@ def _set_omp_num_threads():
 
 
 def _get_dmatrix(data: RayDMatrix, param: Dict) -> xgb.DMatrix:
+    if QUANTILE_AVAILABLE and isinstance(data, RayQuantileDMatrix):
+        if isinstance(param["data"], list):
+            qdm_param = {
+                "data": concat_dataframes(param["data"]),
+                "label": concat_dataframes(param["label"]),
+                "weight": concat_dataframes(param["weight"]),
+                "feature_weights": concat_dataframes(param["feature_weights"]),
+                "qid": concat_dataframes(param["qid"]),
+                "base_margin": concat_dataframes(param["base_margin"]),
+                "label_lower_bound": concat_dataframes(
+                    param["label_lower_bound"]),
+                "label_upper_bound": concat_dataframes(
+                    param["label_upper_bound"]),
+            }
+            param.update(qdm_param)
+        if data.enable_categorical is not None:
+            param["enable_categorical"] = data.enable_categorical
+        matrix = xgb.QuantileDMatrix(**param)
```

Comment:
Thanks, this is great! Since this block of code is now used in two places, could you separate it out as a helper function?